### PR TITLE
feat(v3-preset): add v3 preset

### PR DESCRIPTION
**Pull request by ElijahKotyluk**
### Install Vuetify 3-alpha

If you haven't yet installed vue-cli 3, first follow the install instructions here: https://github.com/vuejs/vue-cli

Generate a project using vue-cli 3.0:
```bash
vue create my-app
```

When prompted, choose `Vue 3 Preview`:
```console
? Please pick a preset: 
   Default ([Vue 2] babel, eslint) 
❯  Default (Vue 3 Preview) ([Vue 3] babel, eslint) 
   Manually select features 

```

Before installing the vuetify plugin, make sure to commit or stash your changes in case you need to revert the changes.

To install the vuetify plugin simply navigate to your application folder and add Vuetify.
```bash
cd my-app

vue add vuetify
```

Once prompted, choose `V3 (alpha)`:
```console
? Choose a preset: (Use arrow keys)
  Default (recommended) 
  Prototype (rapid development) 
  Configure (advanced) 
> V3 (alpha)
```

If you run into any issues, reach out to us in our [Discord Community server](https://community.vuetifyjs.com).

### CODE REVIEW

File: packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js [hunk 1]
```diff
@@ -2,12 +2,12 @@
 const fs = require('fs')
 const helpers = require('./helpers')
 
-function addDependencies (api) {
+function addDependencies (api, v3) {
   api.extendPackage({
     dependencies: {
-      vuetify: '^2.4.0',
+      vuetify: v3 ? "^3.0.0-alpha" : "^2.4.0", // @TODO Check exact version
```
Comment (KaelWD) [inline packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js:8]:
```suggestion
      vuetify: v3 ? "^3.0.0-alpha.0" : "^2.4.0",
```

File: packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
     },
-  })
+  });
```
Comment (KaelWD) [inline packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js:10]:
```suggestion
  })
```

**Review by KaelWD — CHANGES_REQUESTED** [review]
The template needs to change too.

**Comment by ElijahKotyluk** [comment]
Looking for some feedback, let me know if I'm missing a dependency or have something wrong. I'll get the Vue template files filled out by tomorrow.

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/App.ts.vue [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<template>
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/generator/templates/v3/src/App.ts.vue:1]: Should this have anything in it?

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/plugins/vuetify.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import "@mdi/font/css/materialdesignicons.css";
+import { createVuetify } from "vuetify";
+import { aliases, mdi } from "vuetify/src/iconsets/mdi";
+import { fa } from "vuetify/src/iconsets/fa-svg";
+
+export default createVuetify({
+  // lang: {
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/generator/templates/v3/src/plugins/vuetify.js:7]: remove commented out code

File: packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js [hunk 5]
```diff
@@ -30,26 +31,39 @@ function renderFiles (api, { opts }) {
...
-      './src/components/HelloWorld.vue': `../templates/default/src/components/HelloWorld.${ext}.vue`,
+    if (opts.useV3) {
+      files = {
+        "./src/App.vue": `../templates/v3/src/App.${ext}.vue`,
+        "./src/assets/logo.svg": "../templates/v3/src/assets/logo.svg",
+        "./src/components/HelloWorld.vue": `../templates/v3/src/components/HelloWorld.${ext}.vue`,
+        [api.entryFile]: "../templates/v3/src/main.js",
+      };
+    } else {
+      files = {
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js:47]: I'd move 47/51 into the default `let files` on 34 and remove the else.

**Review by johnleider — CHANGES_REQUESTED** [review]

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/App.ts.vue (hunk 3 above)
Comment (ElijahKotyluk) [reply packages/vue-cli-plugin-vuetify/generator/templates/v3/src/App.ts.vue:1]: I'm going to fill out the component templates today.

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/plugins/vuetify.js (hunk 4 above)
Comment (ElijahKotyluk) [reply packages/vue-cli-plugin-vuetify/generator/templates/v3/src/plugins/vuetify.js:7]: Do we need any of the commented out portion at the moment?

**Review by johnleider — APPROVED** [review]

**Pull request merged**